src/plugin.rs

```diff
@@ -60,9 +63,9 @@ where
     }
 }
 
-impl<R: SeedableEntropySource + 'static> Plugin for EntropyPlugin<R>
+impl<R: SeedableEntropySource + Typed + 'static> Plugin for EntropyPlugin<R>
```

Comment:
You can move the `Typed` trait constraint to be part of the `SeedableEntropySource` constraints in `bevy_prng`. This way, you can reduce some of the noise in declaring `Typed` in various places that depend on `SeedableEntropySource`.

Reply:
Done. One thing that's a little unclear is whether we should put the `Typed` trait bound on `SeedableRng::Seed` on the trait or just `GlobalEntropy`. I moved it to the trait. There are a lot of trait bounds for `R::Seed` in `GlobalEntropy` that seem like they should probably just be on the trait?

Reply:
Huh, looks like this change broke `Seed512` for `Xoshiro512` variants. And maybe, it has been broken for a long while because of the reflect requirements, but I don't think I can newtype my way out of this one in third-party land. RIP.

Reply:
I just moved the `Typed` bound for the `SeedableEntropySource::Seed` to the `Plugin`. You're correct this doesn't solve it, but at least it gets the crate compiling for Bevy 0.15.

Reply:
Well, doing so just brought to light that the `Xoshiro512` PRNG variants never really worked, because the `Seed512` struct would have never worked with the reflection trait constraints. So there's no point in supporting it, since it is a case I literally can't get around as a third-party crate due to orphan rules. For now, just go into the `xoshiro.rs` and delete the three 512 variants, plus clean up wherever they are referenced.

Reply:
Ahhh ok I understand now. I've reverted back to putting `Seed: Typed` on the `SeedableEntropySource` trait`, and remove `Xoshiro512*` and `Seed512`.

One thing we could consider is using [remote reflection](https://github.com/bevyengine/bevy/pull/6042). I'm not sure the full implications, though I know it requires copying the definition of the underlying type which isn't great.

Reply:
I'm not 100% sure that will work in our case for remote reflection, because the `SeedableRng` impl for Xoshiro512 variants work with a concrete type already. So I'm not fully convinced it'll work out. But all of that should be left to a separate PR to explore that. For now, removing what was never working in the first place is not really going to be breaking for anyone else so it is "safe".